opencv_ai/splice/component.h

```diff
@@ -0,0 +1,60 @@
+#ifndef OPENCV_AI_SPLICE_COMPONENT_H_
+#define OPENCV_AI_SPLICE_COMPONENT_H_
+
+#include <boost/optional.hpp>
+#include <boost/regex.hpp>
+
+#include <glog/logging.h>
+
+#include <opencv_ai/splice/context.h>
+
+namespace opencv_ai {
```

Comment:
I don't mind the opencv_ai brand for a namespace, similar in spirit to ``boost::``.  splice conflicts with:

```
/usr/include/x86_64-linux-gnu/bits/fcntl-linux.h:369:16: note: previous definition is here
extern ssize_t splice (int __fdin, __off64_t *__offin, int __fdout,